starter-code/app.js

```diff
@@ -26,7 +31,21 @@ app.use(bodyParser.urlencoded({ extended: false }));
 // Authentication
 app.use(cookieParser());
 
+// Cookies
+app.use(session({
+  secret: "basic-auth-secret",
+  cookie: { maxAge: 60000 },
+  store: new MongoStore({
+    mongooseConnection: mongoose.connection,
+    ttl: 24 * 60 * 60 // 1 day
+  }),
+  resave: true,
+  saveUninitialized: true
+}));
+
 // Routes
+app.use('/', index);
```

Comment:
you should have used two routers
- one for pages only accessible to non-authenticated users  (when `req.session.currentUser` is NOT defined)
- another one for users already logged in (when `req.session.currentUser` is defined)

if you had split in two routers, then at the top of each router file you could have defined a simple middleware to run before the routes, allowing you to write the the  `if (req.session.currentUser)` guard only once per router